daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala

```diff
@@ -1133,39 +1104,30 @@ object JsonLedgerClient {
         "payload" -> args.payload,
         "choice" -> args.choice.toJson,
         "argument" -> args.argument)
-    implicit val createAndExerciseReader: RootJsonReader[CreateAndExerciseResponse] = v => {
-      v.asJsObject.getFields("result") match {
-        case Seq(result) =>
-          result.asJsObject.getFields("exerciseResult", "events") match {
-            case Seq(result, events) =>
-              events match {
-                case JsArray(events) if events.size >= 1 =>
-                  events.head.asJsObject.getFields("created") match {
-                    case Seq(created) =>
-                      created.asJsObject.getFields("contractId") match {
-                        case Seq(JsString(cid)) => CreateAndExerciseResponse(cid, result)
-                        case _ =>
-                          deserializationError(s"Could not parse CreateAndExerciseResponse: $v")
-                      }
-                    case _ => deserializationError(s"Could not parse CreateAndExerciseResponse: $v")
+    implicit val createAndExerciseReader: RootJsonReader[CreateAndExerciseResponse] = v =>
+      v.asJsObject.getFields("exerciseResult", "events") match {
+        case Seq(result, events) =>
+          events match {
+            case JsArray(events) if events.size >= 1 =>
+              events.head.asJsObject.getFields("created") match {
```

Comment:
Maybe we should ban `head` 🔨 (I know just a reindent but...)

```suggestion
            case JsArray(Seq(event, _*)) =>
              event.asJsObject.getFields("created") match {
```

Reply:
good catch, I didn’t pay attention to that.